src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -232,32 +234,48 @@ private bool StartCore(ProcessStartInfo startInfo)
                 {
                     throw new InvalidOperationException(SR.CantRedirectStreams);
                 }
+            }
 
-                const string ShellPath = "/bin/sh";
+            int childPid = -1, stdinFd = -1, stdoutFd = -1, stderrFd = -1, result = -1;
+            string[] envp = CreateEnvp(startInfo);
+            string cwd = !string.IsNullOrWhiteSpace(startInfo.WorkingDirectory) ? startInfo.WorkingDirectory : null;
 
-                filename = ShellPath;
-                argv = new string[3] { ShellPath, "-c", startInfo.FileName + " " + startInfo.Arguments};
-            }
-            else
+            filename = ResolvePath(startInfo.FileName);
```

Comment:
>filename = ResolvePath(startInfo.FileName); [](start = 12, length = 43)

could you please add some comment telling even when having ShellExecute option, we still try first launching the FileName and if didn't work we'll try the shell tools to launch it.